baseband_tasks/tests/test_sampling.py

```diff
@@ -681,6 +686,18 @@ def test_compare_with_shift_and_resample(self):
         resampled = resampler.read(20)
         assert_allclose(shifted, resampled)
 
-    def test_error_on_non_integer_shift(self):
-        with pytest.raises(TypeError, match='cast'):
-            SampleShift(self.ih, np.array([1., 2., 3.5]))
+    def test_non_integer_shift(self):
+        shifter1 = SampleShift(self.ih, np.array([1., 2., 3.25]))
+        comparison = SampleShift(self.ih, np.array([1, 2, 3]))
```

Comment:
Should we give a shift with its fractional part >= 0.5? Just to complete the case.

Reply:
Plus, should we give a noninteger shift with the other axis, as the reason above, complete the test cases.

Reply:
Sure, done!